src/Core/Content/Product/DataAbstractionLayer/ProductStreamUpdater.php

```diff
@@ -83,6 +83,7 @@ public function handle(EntityIndexingMessage $message): void
         if ($criteria === null) {
             return;
         }
+        $criteria->addState(Criteria::STATE_ELASTICSEARCH_AWARE);
```

Comment:
was there a reason we did not rely on elastic during the indexing? We know that indexing of product stream can take some time, using ES could speed that up as well

Reply:
there're no reason I think, the counter reason is the ES was introduced solely for searching/product listing and it was not utilized in many places.

But there it makes a lot of sense here to use elasticsearch to boost this query 👍🏽